[libgnutls] Static use depends on -ldl, not carried in pc file

**Describe the bug**
While trying to update port libmicrohttpd, I stumbled over unresolved symbols when using static build of port libgnutls in config.log:

~~~
configure:24362: checking whether GnuTLS is usable
configure:24385: gcc -o conftest -I/home/kpa/Projekte/installed/x64-linux/debug/lib/pkgconfig/../../../include -fPIC -g -fno-strict-aliasing  -L/home/kpa/Projekte/installed/x64-linux/debug/lib  conftest.c  -L/home/kpa/Projekte/installed/x64-linux/debug/lib/pkgconfig/../../lib -lgnutls -pthread -ltspi -lgmp -latomic -lhogweed -lgmp -lnettle -ltasn1 -lidn2 >&5
/home/kpa/Projekte/installed/x64-linux/debug/lib/libgnutls.a(tpm.o): In function `check_init':
/home/kpa/Projekte/buildtrees/libgnutls/x64-linux-dbg/lib/.././../src/3.6.15-4050d06d46.clean/lib/tpm.c:123: undefined reference to `dlopen'
/home/kpa/Projekte/buildtrees/libgnutls/x64-linux-dbg/lib/.././../src/3.6.15-4050d06d46.clean/lib/tpm.c:129: undefined reference to `dlsym'
/home/kpa/Projekte/buildtrees/libgnutls/x64-linux-dbg/lib/.././../src/3.6.15-4050d06d46.clean/lib/tpm.c:129: undefined reference to `dlclose'
...
collect2: error: ld returned 1 exit status
...
configure:24397: result: no
configure:24407: WARNING: pkg-config reports that GnuTLS is present, but GnuTLS can't be used
~~~

I'm not sure if this is desired for the static build, or if it is just a bug in the `.pc` files.

**Environment**
- OS: linux
- Triplet: default (x64-linux)

**To Reproduce**
1. Add libgnutls dependency to libmicrohttpd
2. Add `--with-https` to OPTIONS for vcpkg_configure_make in libmicrohttpd portfile.
3. `./vcpkg install libmicrohttpd`
4. Inspect config.log


**Expected behavior**
No undefined references when consuming libgnutls via .pc files in a static build

--- a/ports/libgnutls/portfile.cmake
+++ b/ports/libgnutls/portfile.cmake
@@ -35,6 +35,7 @@ vcpkg_configure_make(
         --disable-libdane
         --with-included-unistring
         --without-p11-kit
+        --without-tpm
         ${OPENSSL_COMPATIBILITY}
         "LDFLAGS=${LDFLAGS}"
 )

--- a/ports/libgnutls/portfile.cmake
+++ b/ports/libgnutls/portfile.cmake
@@ -10,7 +10,7 @@ vcpkg_download_distfile(ARCHIVE
 
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    ARCHIVE "${ARCHIVE}"
     REF ${GNUTLS_VERSION}
 )
 
@@ -25,7 +25,7 @@ if ("openssl" IN_LIST FEATURES)
 endif()
 
 vcpkg_configure_make(
-    SOURCE_PATH ${SOURCE_PATH}
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         --disable-doc
         --disable-silent-rules
@@ -44,5 +44,5 @@ vcpkg_install_make()
 vcpkg_fixup_pkgconfig()
 vcpkg_copy_pdbs()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/libgnutls/vcpkg.json
+++ b/ports/libgnutls/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libgnutls",
   "version": "3.6.15",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A secure communications library implementing the SSL, TLS and DTLS protocols",
   "homepage": "https://www.gnutls.org/",
   "supports": "!windows",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3134,7 +3134,7 @@
     },
     "libgnutls": {
       "baseline": "3.6.15",
-      "port-version": 1
+      "port-version": 2
     },
     "libgo": {
       "baseline": "3.1-1",

--- a/versions/l-/libgnutls.json
+++ b/versions/l-/libgnutls.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c03a1c452fa39d1b6d884aa3ef12c0c98a11f0a3",
+      "version": "3.6.15",
+      "port-version": 2
+    },
+    {
       "git-tree": "09f2d8c5e4e07d2076324767d251fef3bc4acb8c",
       "version": "3.6.15",
       "port-version": 1

